core/src/main/java/com/scalar/db/transaction/consensuscommit/ConsensusCommitAdmin.java

```diff
@@ -399,4 +530,34 @@ private void checkNamespace(String namespace) {
           CoreError.CONSENSUS_COMMIT_COORDINATOR_NAMESPACE_SPECIFIED.buildMessage(namespace));
     }
   }
+
+  private boolean isTransactionMetadataDecouplingEnabled(Map<String, String> options) {
+    return options.containsKey(TRANSACTION_METADATA_DECOUPLING)
+        && options.get(TRANSACTION_METADATA_DECOUPLING).equalsIgnoreCase("true");
+  }
+
+  private void throwIfTransactionMetadataDecouplingNotSupportedStorage(String namespace)
+      throws ExecutionException {
+    StorageInfo storageInfo = admin.getStorageInfo(namespace);
+    if ((storageInfo.getMutationAtomicityUnit() != StorageInfo.MutationAtomicityUnit.STORAGE
+        && storageInfo.getMutationAtomicityUnit() != StorageInfo.MutationAtomicityUnit.NAMESPACE)) {
+      throw new IllegalArgumentException(
+          CoreError.CONSENSUS_COMMIT_TRANSACTION_METADATA_DECOUPLING_NOT_SUPPORTED_STORAGE
+              .buildMessage(storageInfo.getStorageName()));
+    }
+    if (!storageInfo.isConsistentVirtualTableReadGuaranteed()) {
+      throw new IllegalArgumentException(
+          CoreError.CONSENSUS_COMMIT_TRANSACTION_METADATA_CONSISTENT_READS_NOT_GUARANTEED_STORAGE
+              .buildMessage(storageInfo.getStorageName()));
+    }
+  }
+
+  private void throwIfTransactionMetadataDecouplingTable(
```

Comment:
```suggestion
  private void throwIfTransactionMetadataDecouplingApplied(
```
I feel `Table` can be omitted since it's specified in the argument.